src/protocol/binary.rs

```diff
@@ -217,18 +270,50 @@ impl BinaryProtocol {
     }
 
     pub(super) fn delete(&mut self, key: &str) -> Result<bool, MemcacheError> {
-        check_key_len(key)?;
-        let request_header = PacketHeader {
-            magic: Magic::Request as u8,
-            opcode: Opcode::Delete as u8,
-            key_length: key.len() as u16,
-            total_body_length: key.len() as u32,
-            ..Default::default()
-        };
-        request_header.write(&mut self.stream)?;
-        self.stream.write_all(key.as_bytes())?;
+        Ok(self.deletes(&[key])?[0])
+    }
+
+    pub(super) fn deletes<K: AsRef<str>>(&mut self, keys: &[K]) -> Result<Vec<bool>, MemcacheError> {
+        for k in keys.iter() {
+            check_key_len(k.as_ref())?;
+        }
+
+        for k in keys.iter() {
+            let key = k.as_ref();
+            let request_header = PacketHeader {
+                magic: Magic::Request as u8,
+                opcode: Opcode::Delete as u8,
+                key_length: key.len() as u16,
+                total_body_length: key.len() as u32,
+                ..Default::default()
+            };
+            request_header.write(&mut self.stream)?;
+            self.stream.write_all(key.as_bytes())?;
+        }
+        // Flush now that all the requests have been written.
         self.stream.flush()?;
-        return binary_packet::parse_delete_response(&mut self.stream);
+
+        // Receive all the responses. If there were errors, return the first.
+        let mut final_result = Ok(Vec::with_capacity(keys.len()));
+
+        for _ in 0..keys.len() {
+            match binary_packet::parse_delete_response(&mut self.stream) {
+                Ok(deleted) => {
+                    if let Ok(deleted_list) = &mut final_result {
+                        deleted_list.push(deleted);
+                    }
+                }
+                Err(MemcacheError::CommandError(e)) => {
```

Comment:
You can use the function [here](https://github.com/aisk/rust-memcache/pull/108/files#diff-9a269209bed069ffb36d5565a91c9840R77) to determine whether the error is recoverable. Please check if you can utilize it. Feel free to make a copy of it to your branch.

Reply:
Thanks!